File: templates/backend/controller.ts
// Template: NestJS controller
// Location: src/modules/<module-name>/<module-name>.controller.ts
// Note: JwtAuthGuard is applied globally — all routes protected by default.
// Use @Public() decorator to opt a route out of auth.

import { Controller, Get, Post, Put, Delete, Param, Body, HttpCode, HttpStatus } from '@nestjs/common'
import { ModuleNameService } from './module-name.service'
import { CreateModuleNameDto } from './dto/create-module-name.dto'
import { UpdateModuleNameDto } from './dto/update-module-name.dto'

@Controller('module-name')
export class ModuleNameController {
  constructor(private readonly moduleNameService: ModuleNameService) {}

  @Get()
  findAll() {
    return this.moduleNameService.findAll()
  }

  @Get(':id')
  findOne(@Param('id') id: string) {
    return this.moduleNameService.findOne(id)
  }

  @Post()
  create(@Body() dto: CreateModuleNameDto) {
    return this.moduleNameService.create(dto)
  }

  @Put(':id')
  update(@Param('id') id: string, @Body() dto: UpdateModuleNameDto) {
    return this.moduleNameService.update(id, dto)
  }

  @Delete(':id')
  @HttpCode(HttpStatus.NO_CONTENT)
  remove(@Param('id') id: string) {
    return this.moduleNameService.remove(id)
  }
}

File: dotnet/templates/Controller.cs
// Template: ASP.NET Core ApiController
// Location: src/Modules/<ModuleName>/<ModuleName>Controller.cs

using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

namespace YourApp.Modules.ModuleName;

/// <summary>ModuleName management endpoints</summary>
[ApiController]
[Route("api/[controller]")]
[Authorize]
public class ModuleNameController : ControllerBase
{
    private readonly IModuleNameService _service;

    public ModuleNameController(IModuleNameService service)
    {
        _service = service;
    }

    /// <summary>List all ModuleName items</summary>
    [HttpGet]
    [SwaggerOperation(Summary = "List all ModuleName items", Tags = new[] { "ModuleName" })]
    [ProducesResponseType(typeof(IEnumerable<ModuleName>), StatusCodes.Status200OK)]
    public async Task<IActionResult> FindAll()
        => Ok(await _service.FindAllAsync());

    /// <summary>Get one ModuleName item by ID</summary>
    [HttpGet("{id}")]
    [SwaggerOperation(Summary = "Get one ModuleName item", Tags = new[] { "ModuleName" })]
    [ProducesResponseType(typeof(ModuleName), StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    public async Task<IActionResult> FindOne(string id)
        => Ok(await _service.FindOneAsync(id));

    /// <summary>Create a new ModuleName item</summary>
    [HttpPost]
    [SwaggerOperation(Summary = "Create ModuleName item", Tags = new[] { "ModuleName" })]
    [ProducesResponseType(typeof(ModuleName), StatusCodes.Status201Created)]
    [ProducesResponseType(StatusCodes.Status400BadRequest)]
    public async Task<IActionResult> Create([FromBody] CreateModuleNameDto dto)
    {
        var item = await _service.CreateAsync(dto);
        return CreatedAtAction(nameof(FindOne), new { id = item.Id }, item);
    }

    /// <summary>Update an existing ModuleName item</summary>
    [HttpPut("{id}")]
    [SwaggerOperation(Summary = "Update ModuleName item", Tags = new[] { "ModuleName" })]
    [ProducesResponseType(typeof(ModuleName), StatusCodes.Status200OK)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    public async Task<IActionResult> Update(string id, [FromBody] UpdateModuleNameDto dto)
        => Ok(await _service.UpdateAsync(id, dto));

    /// <summary>Delete a ModuleName item</summary>
    [HttpDelete("{id}")]
    [SwaggerOperation(Summary = "Delete ModuleName item", Tags = new[] { "ModuleName" })]
    [ProducesResponseType(StatusCodes.Status204NoContent)]
    [ProducesResponseType(StatusCodes.Status404NotFound)]
    public async Task<IActionResult> Remove(string id)
    {
        await _service.RemoveAsync(id);
        return NoContent();
    }
}
